src/pages/components/DataAndTimeSelector.tsx

```diff
@@ -23,27 +22,34 @@ export function DataAndTimeSelector({
       setTimestamp(timeSelected)
   }, [timeSelected])
 
+  useEffect(() => {
+    if (timeSelected != null && !timeSelected.isSame(timestamp)) setTimeSelected(timestamp)
+  }, [timestamp])
+
+  const width = showTimePicker ? '50%' : '100%'
+
   return (
     <>
       <DatePicker
-        sx={{ width: INPUT_SIZE }}
+        sx={{ width }}
         value={timeSelected}
         onChange={(ts) => setTimeSelected(ts)}
         format="DD/MM/YYYY"
-        label={TEXTS.choose_datetime}
+        label={TEXTS.choose_date}
         disableFuture
       />
       {showTimePicker && (
         <TimePicker
```

Comment:
There is no space between those input fields . Please, add some space.
![image](https://github.com/hasadna/open-bus-map-search/assets/76536506/dc596be8-05df-46bf-bcdc-1d83f44e0c97)

Reply:
I refactor the DateAndTimeSelector to two different components and fix the problem.